guide/kong_integration.md

```diff
@@ -0,0 +1,195 @@
+```
+
+Verify that authentication works
+
+```
+$ curl localhost:8000/function/func_echoit -d 'hello world'
+{"message":"Unauthorized"}
+
+$ curl localhost:8000/function/func_echoit -d 'hello world' \
+    -H 'Authorization: Basic xxxxxx'
+{"message":"Invalid authentication credentials"}
+
+$ echo -n aladdin:OpenSesame | base64
+YWxhZGRpbjpPcGVuU2VzYW1l
+
+$ curl localhost:8000/function/func_echoit -d 'hello world' \
+    -H 'Authorization: Basic YWxhZGRpbjpPcGVuU2VzYW1l'
+hello world
+```
+
+Now lets expose the /ui directory so we can securely use the web GUI
+
+```
+$ curl -i -X POST \
+    --url http://localhost:8001/apis/ \
+    --data 'name=ui' \
+    --data 'uris=/ui' \
+    --data 'upstream_url=http://gateway:8080/ui'
+```
+
+Additionally we need to expose /system/functions since the UI makes Ajax requests to it
+
+```
+$ curl -i -X POST \
+    --url http://localhost:8001/apis/ \
+    --data 'name=system-functions' \
+    --data 'uris=/system/functions' \
+    --data 'upstream_url=http://gateway:8080/system/functions'
+```
+
+Verify that the UI is secure
+
+```
+$ curl -i localhost:8000/ui/ \
+    -H 'Authorization: Basic YWxhZGRpbjpPcGVuU2VzYW1l'
+
+HTTP/1.1 200 OK
+Content-Type: text/html; charset=utf-8
+...
+```
+
+Now visit http://localhost:8000/ui/ in your browser where you will be asked for credentials.
+
+### Add SSL
+
+Basic authentication does not protect from man in the middle attacks, so lets add SSL to encrypt the communication.
+
+Create a cert. Here in the demo, we are creating selfsigned certs, but in production you should skip this step and use your existing certificates (or get some from Lets Encrypt).
+```
+$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+  -keyout /tmp/selfsigned.key -out /tmp/selfsigned.pem \
+  -subj "/C=US/ST=CA/L=L/O=OrgName/OU=IT Department/CN=example.com"
+```
+
+Add cert to Kong
+
+```
+$ curl -i -X POST http://localhost:8001/certificates \
+    -F "cert=@/tmp/selfsigned.pem" \
+    -F "key=@/tmp/selfsigned.key" \
+    -F "snis=example.com"
+
+HTTP/1.1 201 Created
+...
+```
+
+Use the cert to secure OpenFaaS
+
+```
+$ curl -i -X POST http://localhost:8001/apis \
+    -d "name=ssl-api" \
+    -d "upstream_url=http://gateway:8080" \
+    -d "hosts=example.com"
+HTTP/1.1 201 Created
+...
+```
+
+Verify that the cert is now in use. Note the '-k' parameter is just here to work around the fact that we are using self signed certs.
+```
+$ curl -k https://localhost:8443/function/func_echoit \
+  -d 'hello world' -H 'Authorization: Basic YWxhZGRpbjpPcGVuU2VzYW1l'
+hello world
+```
+
+## Configure your firewall
+
+Between OpenFaaS and Kong a lot of ports are exposed on your host machine. In the end it is best to only expose either 8000 or 8443 out of your network depending if you added SSL or not.
+
+Another option is to expose both and enable [https_only](https://getkong.org/docs/0.11.x/proxy/#the-https_only-property) which is used to notify clients to upgrade to https from http.
```

Comment:
If you change this URL to https://getkong.org/docs/latest/proxy/#the-https_only-property it'll automatically point to the latest release of Kong CE.